### ISSUE TITLE
elastix_napari plugin is incompatible with magicgui=0.4.0

### ISSUE THREAD

**Issue by GenevieveBuckley** [SuperElastix/elastix_napari#19]
This napari plugin is incompatible with the dependency `magicgui` version 0.4.0. This prevents users of the `elastix_napari` plugin from using any of the advanced registration parameters.

## Steps to reproduce the error
1. Create a new elastix environment
```
conda create -n elastix python=3.9 pip
conda activate elastix
conda install napari -c conda-forge
pip install elastix_napari
```
2. Launch napari, and start the `elastix_napari` plugin.
3. Click the "Advanced" checkbox. Observe that the dock widget does not open the advanced parameters. An error traceback message is produced.

## Error message
```python-traceback
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
File ~\.conda\envs\elastix\lib\site-packages\magicgui\widgets\_bases\value_widget.py:57, in ValueWidget._on_value_change(self=CheckBox(value=True, annotation=, name='advanced'), value=True)
     55 if value is self.null_value and not self._nullable:
     56     return
---> 57 self.changed.emit(value)
        value = True
        self.changed = , name='advanced')>
        self = CheckBox(value=True, annotation=, name='advanced')

File ~\.conda\envs\elastix\lib\site-packages\psygnal\_signal.py:681, in psygnal._signal.SignalInstance.emit()

File ~\.conda\envs\elastix\lib\site-packages\psygnal\_signal.py:723, in psygnal._signal.SignalInstance._run_emit_loop()

File ~\.conda\envs\elastix\lib\site-packages\psygnal\_signal.py:724, in psygnal._signal.SignalInstance._run_emit_loop()

File ~\.conda\envs\elastix\lib\site-packages\psygnal\_signal.py:744, in psygnal._signal.SignalInstance._run_emit_loop()

File ~\.conda\envs\elastix\lib\site-packages\elastix_napari\elastix_registration.py:56, in on_init..toggle_advanced_widget(event=True)
     52 else:
     53     for x in ['metric', 'init_trans', 'resolutions',
     54               'max_iterations', 'nr_spatial_samples',
     55               'max_step_length', 'fixed_ps', 'moving_ps']:
---> 56         setattr(getattr(widget, x), 'visible', event.value)
        x = 'metric'
        widget =  napari.layers.image.image.Image>
        event = True

AttributeError: 'bool' object has no attribute 'value'

```

## Solutions
For a quick fix, users can downgrade `magicgui` to version 0.3.7. I do not observe the same problem with this version of magicgui.

As an interim solution, you could also pin the magicgui requirement more tightly, eg: `magicgui>=0.2.6,<0.4.0` and push another release of the plugin to PyPI.

For the proper fix, I'm not exactly sure what the breaking change in magicgui was. There is only one deprecation mentioned in the last release, and it's not obvious to me if that's the same thing we've seen here or not (I think it's different?) https://github.com/napari/magicgui/releases  Perhaps @tlambert03 can provide some more insight.

**Comment by GenevieveBuckley** [comment]
Oh hang on, yes, the deprecation mentioned is the source of the incompatibility. There's even quite a nice warning message when you run the plugin with magicgui 0.3.7

```
magicgui 0.4.0 will change the way that callbacks are called
Instead of a single `Event` instance, with an `event.value` attribute,
callabacks will recieve the value(s) directly:

@advanced.changed.connect
def my_callback(*args):
    # *args are the value(s) themselves!
```

I think if the callback function signature is type annotated, eg: `def callback(x: bool): ...` then that might make the problem go away. It's worth a try.

**Comment by GenevieveBuckley** [comment]
...scratch that idea, it is already type-annotated:
https://github.com/SuperElastix/elastix_napari/blob/5180be2ccd3ca5294a4bab149d7a95fc9e9e022e/elastix_napari/elastix_registration.py#L102

Perhaps it's this line that is the problem. That could fit with the warning message about no longer having a `.value` attribute.
https://github.com/SuperElastix/elastix_napari/blob/5180be2ccd3ca5294a4bab149d7a95fc9e9e022e/elastix_napari/elastix_registration.py#L46

**Comment by tlambert03** [comment]
See https://napari.org/magicgui/api/v0_3_0.html

**Comment by ntatsisk** [comment]
Together with @ViktorvdValk and @N-Dekker, we just released a new version `0.1.8` that fixes the current issues.

@GenevieveBuckley Do you mind taking a look if it also works for you now?

**Comment by GenevieveBuckley** [comment]
Yes, that's fixed it!

<img width="1386" alt="Screen Shot 2023-03-31 at 4 27 46 pm" src="https://user-images.githubusercontent.com/30920819/229031049-a593279f-d268-4dfe-942b-61d71ada95e4.png">
